src/frontend/SignatureMismatch.ml

```diff
@@ -70,9 +70,16 @@ and details =
   | ReturnTypeMismatch of UnsizedType.returntype * UnsizedType.returntype
   | InputMismatch of function_mismatch
 
+and promotions =
+  | None
+  | IntToRealPromotion
+  | IntToComplexPromotion
+  | RealToComplexPromotion
+
 and function_mismatch =
   | ArgError of int * type_mismatch
   | ArgNumMismatch of int * int
+  | PromotionConflict of promotions list * promotions list
```

Comment:
I don't think `PromotionConflict` belongs here. It's not the same kind of error as the others.